source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -486,12 +486,21 @@ Http::FilterTrailersStatus JsonTranscoderFilter::encodeTrailers(Http::HeaderMap&
   // so there is no need to copy headers from one to the other.
   bool is_trailers_only_response = response_headers_ == &trailers;
 
-  if (!grpc_status || grpc_status.value() == Grpc::Status::GrpcStatus::InvalidCode) {
+  // If grpc-status is greater than Grpc::Status::MaximumValid, that is not invalid because if may
+  // be user defined grpc-status. so that must handled as 200 OK
+  bool grpc_status_greater_maximum = grpc_status.value() > Grpc::Status::MaximumValid;
+
+  if (!grpc_status || grpc_status.value() == Grpc::Status::GrpcStatusMapping::InvalidCode) {
     response_headers_->Status()->value(enumToInt(Http::Code::ServiceUnavailable));
   } else {
-    response_headers_->Status()->value(Grpc::Utility::grpcToHttpStatus(grpc_status.value()));
+    if (grpc_status_greater_maximum) {
+      response_headers_->Status()->value(200);
```

Comment:
because you changed grpcToHttpStatus to default to 500, I don't think you need special handling here.